src/react/components/shell/mainContentRouter.tsx

```diff
@@ -15,7 +16,27 @@ import ProjectSettingsPage from "../pages/projectSettings/projectSettingsPage";
  * @name - Main Content Router
  * @description - Controls main content pane based on route
  */
+
+interface IProjectId {
+    projectId: string;
+}
+
 export function MainContentRouter() {
+    const id = getPathParams(window.location.pathname, "projectId") as IProjectId;
```

Comment:
cant we get the id from the Redux store instead of the url

Reply:
I think, it is faster to get it from path directly neither wait for response from Redux. Also Redux reload many times... 
![image](https://user-images.githubusercontent.com/64093224/83298285-fd08a000-a1a8-11ea-8cf3-17de9681f84a.png)

Reply:
it seems redundant to add a projectId state to this component when it's already in the redux store. Usually you want to lift state. But maybe not necessary in this case